main.py

```diff
@@ -35,17 +35,21 @@
 
     args = parser.parse_args()
     # Load integrals
-    n_ord, E0, d_one_e_integral, d_two_e_integral = load_integrals(args.fcidump_path)
+    comm = MPI.COMM_WORLD
+    n_ord, E0, d_one_e_integral, d_two_e_integral = load_integrals(
+        args.fcidump_path, comm
+    )
```

Comment:
There's no good reason to move `comm` inside the function, if only rank 0 does something. You can move the if rank block here, and avoid all those indentation changes to load_integrals.